app/controllers/lti_deployments_controller.rb

```diff
@@ -180,7 +182,17 @@ def check_host
   end
 
   def create_course
-    new_course = Course.find_or_initialize_by(name: params['name'])
+    if LtiConfig.respond_to?(:allowed_to_create_course?) && !LtiConfig.allowed_to_create_course?(record)
```

Comment:
@donny-wong this is interesting. My intention here was that if the method was not defined at all, this would mean there is no filter, so all courses can trigger course creation. I think that's the correct default behaviour, since I view the "whitelist" as an add-on. Does that make sense?